integration/popular_docker_hub_images/popular_docker_images.bats

```diff
@@ -35,343 +37,433 @@ setup() {
-	docker run --rm --runtime=$RUNTIME -i telegraf sh -c "telegraf config > telegraf.conf; ls telegraf.conf"
+	image="telegraf"
+	docker run --rm --runtime=$RUNTIME -i $image sh -c "telegraf config > telegraf.conf; ls telegraf.conf"
 }
 
 @test "[java application] run a tomcat container" {
-	docker run --rm --runtime=$RUNTIME -i tomcat sh -c "echo 'public class HW{public static void main(String[]a){System.out.println(\"HelloWorld\");}}' > HW.java | ls -l /usr/local/tomcat/HW.java"
+	image="tomcat"
+	docker run --rm --runtime=$RUNTIME -i $image sh -c "echo 'public class HW{public static void main(String[]a){System.out.println(\"HelloWorld\");}}' > HW.java | ls -l /usr/local/tomcat/HW.java"
 }
 
 @test "[java application] run tomee container" {
-	docker run --rm --runtime=$RUNTIME -i tomee bash -c "echo -e 'public class CL{public static void main(String[]a){System.out.println(\"KataContainers\");}}' > CL.java"
+	image="tomee"
+	docker run --rm --runtime=$RUNTIME -i $image bash -c "echo -e 'public class CL{public static void main(String[]a){System.out.println(\"KataContainers\");}}' > CL.java"
 }
 
 @test "[display version] run an instance in a traefik container" {
-	if docker run --rm --runtime=$RUNTIME -i traefik traefik --version | grep "EXEC spawning"; then false; else true; fi
+	image="traefik"
+	if docker run --rm --runtime=$RUNTIME -i $image traefik --version | grep "EXEC spawning"; then false; else true; fi
 }
 
 @test "[run application] run an instance in an ubuntu debootstrap container" {
-	docker run --rm --runtime=$RUNTIME -i ubuntu-debootstrap sh -c 'if [ -f /etc/bash.bashrc ]; then echo "/etc/bash.bashrc exists"; fi'
+	image="ubuntu-debootstrap"
+	docker run --rm --runtime=$RUNTIME -i $image sh -c 'if [ -f /etc/bash.bashrc ]; then echo "/etc/bash.bashrc exists"; fi'
 }
 
 @test "[run application] search nano in an ubuntu upstart container" {
-	docker run --rm --runtime=$RUNTIME -i ubuntu-upstart bash -c "apt-cache search nano"
+	image="ubuntu-upstart"
+	docker run --rm --runtime=$RUNTIME -i $image bash -c "apt-cache search nano"
 }
 
 @test "[run application] start server in a vault container" {
-	docker run --rm --runtime=$RUNTIME -i -e 'VAULT_DEV_ROOT_TOKEN_ID=mytest' vault timeout -t 10 vault server -dev
+	image="vault"
+	docker run --rm --runtime=$RUNTIME -i -e 'VAULT_DEV_ROOT_TOKEN_ID=mytest' $image timeout -t 10 vault server -dev
 }
 
 @test "[perl application] start wordpress container" {
-	if docker run --rm --runtime=$RUNTIME -i wordpress perl -e 'print "test\n"' | grep LANG; then false; else true; fi
+	image="wordpress"
+	if docker run --rm --runtime=$RUNTIME -i $image perl -e 'print "test\n"' | grep LANG; then false; else true; fi
 }
 
 @test "[run application] start zookeeper container" {
-	docker run --rm --runtime=$RUNTIME -i zookeeper zkServer.sh start
+	image="zookeeper"
+	docker run --rm --runtime=$RUNTIME -i $image zkServer.sh start
 }
 
 teardown() {
 	clean_env
+	docker rmi $image
```

Comment:
I think the idea of dropping/removing/cleaning up the images after each test is probably a sound one. It might end up with us being a bit less efficient on the download side as we might drop some layers that are then used in a later test, but if we need to save the space to run the tests, then so be it.